src/renderer/components/recording-page/recording-visualization.tsx

```diff
@@ -2,226 +2,161 @@ import log from 'electron-log';
-  //     if (waveformCanvases.length) {
-  //       log.silly('Updating waveform image');
-  //       const waveform: HTMLCanvasElement = waveformCanvases[waveformCanvases.length - 1];
-  //       waveformImage.src = waveform.toDataURL();
-  //     }
-  //     const spectrogramCanvases = spectrogramContainer.getElementsByTagName('canvas');
-  //     if (spectrogramCanvases.length) {
-  //       log.silly('Updating spectrogram image');
-  //       const spectrogram: HTMLCanvasElement = spectrogramCanvases[spectrogramCanvases.length - 1];
-  //       spectrogramImage.src = spectrogram.toDataURL();
-  //     }
-  //   }
-  // });
+  const [, , audioInputDeviceId, , ,] = useAudioInputOutputDevices();
+  const fileMonitor = new FileMonitor(basePath);
+
+  function checkSpectrogramPlugin() {
+    const waveSurfer = waveSurferRef.current;
+
+    if (!waveSurfer) {
+      return;
+    }
+
+    if (!('spectrogram' in waveSurfer.getActivePlugins())) {
+      waveSurfer
+        .addPlugin(
+          SpectrogramPlugin.create({
+            container: spectrogramContainerRef.current!,
+            labels: true,
+          }),
+        )
+        .initPlugin('spectrogram');
+    }
+  }
+
+  function loadWavFile(filePathToRead: string) {
+    const waveSurfer = waveSurferRef.current;
+
+    if (!waveSurfer) {
+      return noOp();
+    }
+    let cancelled = false;
+
+    (async (): Promise<void> => {
+      if ((await checkFileExistence(filePathToRead)) === 'file') {
+        const data = await readFile(filePathToRead, true);
+        if (!cancelled) {
+          waveSurfer.loadBlob(new Blob([new Uint8Array(data)]));
+          log.info('Triggered waveform update');
+        }
+      }
+    })();
+
+    return (): void => {
+      cancelled = true;
+    };
+  }
 
   useEffect(() => {
+    // create wavesurfer instance with plugins
     const waveSurfer = WaveSurfer.create({
       interact: false,
       // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
       container: waveformContainerRef.current!,
+      height: 100,
       plugins: [
         SpectrogramPlugin.create({
-          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
           container: spectrogramContainerRef.current!,
           labels: true,
         }),
+        MicrophonePlugin.create({}),
       ],
     });
+
+    // add watchers to the fileMonitor
+    fileMonitor.watch(['add', 'unlink', 'change']);
+    fileMonitor.getSubject().subscribe({
+      next: (x: string[]) => {
+        if (x[1].includes(recItemName) && state !== 'recording') {
+          if (x[0] == 'add') {
+            checkSpectrogramPlugin();
+            const filePathToRead = x[1];
+            loadWavFile(filePathToRead);
+          } else if (x[0] == 'unlink') {
+            alert('File ' + x[1] + ' was deleted.');
+            waveSurfer.empty();
+            waveSurfer.destroyPlugin('spectrogram');
```

Comment:
Why destroy only the `spectrogram` plugin?

Reply:
i destroyed the spectrogram plugin because it does not have any nice empty function lie the waveform does. if you prefer I can change it to destroy the whole thing and re-create it every time it's needed?

Reply:
No, it's not that I prefer it. I guess the next question is: do you need to destroy the spectrogram plugin? Reusing the same instance is almost always preferable.

Reply:
So the spectrogram has very few functions that can be used on an instance. The only way I could find to clear the data out of the spectrogram was to just destroy the entire instance.